template/context.go

```diff
@@ -44,12 +45,13 @@ type Context struct {
 // NewContextFromTask returns a new template context from the data available in
 // task. The provided context can then be used to populate runtime values in a
 // ContainerSpec.
-func NewContextFromTask(t *api.Task) (ctx Context) {
+func NewContextFromTask(t *api.Task, n *api.NodeDescription) (ctx Context) {
```

Comment:
If we make this dependent on `NodeDescription`, we should rename the function to `NewContext`.

Another option would be to keep `NewContextFromTask` as-is, and add a `WithNodeDescription` method that fills in the node-dependent values. This would keep the node description optional and avoid needing to pass it through code paths that don't need it (for example, tests exercising other parts of the templating functionality). However, I'm not sure there's much value in `NodeDescription` being optional if we expect all executors to provide it.

@stevvooe WDYT?